src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/PremisesTest.kt

```diff
@@ -66,6 +67,41 @@ class PremisesTest : IntegrationTestBase() {
       .json(expectedJson)
   }
 
+  @Test
+  fun `Get Premises by ID returns OK with correct body`() {
+    val premises = premisesEntityFactory
+      .withYieldedApArea { apAreaEntityFactory.produceAndPersist() }
+      .withYieldedLocalAuthorityArea { localAuthorityEntityFactory.produceAndPersist() }
+      .withYieldedProbationRegion { probationRegionEntityFactory.produceAndPersist() }
+      .produceAndPersistMultiple(5)
+
+    val premisesToGet = premises[2]
+    val expectedJson = objectMapper.writeValueAsString(premisesEntityToExpectedApiResponse(premises[2]))
+
+    webTestClient.get()
+      .uri("/premises/${premisesToGet.id}")
+      .exchange()
+      .expectStatus()
+      .isOk
+      .expectBody()
+      .json(expectedJson)
+  }
+
+  @Test
+  fun `Get Premises by ID returns Not Found with correct body`() {
+    val idToRequest = UUID.randomUUID().toString()
+
+    webTestClient.get()
+      .uri("/premises/$idToRequest")
```

Comment:
Is it worth testing to see the Content-Type header is `application/problem+json`?